patches/minecraft/net/minecraft/item/Item.java.patch

```diff
@@ -791,13 +791,21 @@
 +     * @param armor the armor itemstack
 +     */
 +    public void onHorseArmorTick(World world, net.minecraft.entity.EntityLiving horse, ItemStack armor) {}
++    
++    /**
++     * @return This Item's TEISR, this method should NOT return a new instance.  Must not be null.
++     */
++    public net.minecraft.client.renderer.tileentity.TileEntityItemStackRenderer getTEISR()
```

Comment:
You should name this method `getSpecialRenderer()` to make it more clear that users may draw models or call tile entity renderers in this method.

Reply:
Users should not draw models or call tile entity renderers in this method.  They should be returning a (preferably static) TileEntityItemStackRenderer instance, which will do those things.

Reply:
I meant that they can do that in the special renderer itself

Reply:
I agree with the name change, it should also be marked as `@SideOnly (Side.CLIENT)`

Reply:
`getTileEntityItemStackRenderer` would be fine by me. I don't think the abbreviation is appropriate, it's way too specific to be abbreviated casually in most contexts, and definitely not well known enough to use it formally here.